Update Volume Snapshot Controller to 7.0.2

jess-sol opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md and developer guide https://git.k8s.io/community/contributors/devel/development.md
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Kubernetes v1.29 installs snapshot controller v7.0.2, which requires an updated ClusterRole to work.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #11037

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
Update Snapshot Controller to 7.0.2 for all supported Kubernetes versions.
```

yankay commented:
/ok-to-test

mzaian reviewed (changes requested):
Hello @jess-sol 

Thank you and could you please make the changes permanent and update the controller version for all the corresponding kubernetes versions here: https://github.com/kubernetes-sigs/kubespray/blob/master/roles/kubespray-defaults/defaults/main/download.yml#L360-L363

Also it's a good opportunity to go up to the latest version https://github.com/kubernetes-csi/external-snapshotter/releases/tag/v7.0.1

jess-sol commented:
Hey @mzaian, I've gone ahead and updated all supported versions of k8s to use external snapshotter 7.0.2, and updated the deployment manifests to parallel those available in the external-snapshotter repo for version 7.0.2. I tested a single deployment with k8s 1.29, and it looks fine. Let me know if anything looks amiss. Thanks!

mzaian commented:
> Hey @mzaian, I've gone ahead and updated all supported versions of k8s to use external snapshotter 7.0.2, and updated the deployment manifests to parallel those available in the external-snapshotter repo for version 7.0.2. I tested a single deployment with k8s 1.29, and it looks fine. Let me know if anything looks amiss. Thanks!

Thanks a lot @jess-sol 

/cc @VannTen @yankay

mzaian commented:
@jess-sol 
Maybe change the PR title and please add a release note about the changes.

jess-sol commented:
Okay should be good now @mzaian

mzaian commented:
> Okay should be good now @mzaian

Thank you
/lgtm
